src/types/widget.ts

```diff
@@ -0,0 +1,23 @@
+import {
+  Helper,
+  HelperState,
+  SearchResults,
+  InstantSearch,
+  SearchParameters,
+} from './instantsearch';
+
+interface WidgetArgs {
+  helper: Helper;
+  state?: HelperState;
+  results?: SearchResults;
+  instantSearchInstance: InstantSearch;
+}
+
+export interface Widget {
+  init(options: WidgetArgs): void;
+  render(options: WidgetArgs): void;
```

Comment:
The `init` & `render` methods don't share the same arguments e.g. `searchMetaData`, `results`.

https://github.com/algolia/instantsearch.js/blob/3de59a32a08df71e0d4ba25af060560d4d6495ec/src/lib/InstantSearch.js#L416-L423

https://github.com/algolia/instantsearch.js/blob/3de59a32a08df71e0d4ba25af060560d4d6495ec/src/lib/InstantSearch.js#L392-L402